scripts/flash_algo.py

```diff
@@ -293,13 +293,25 @@ def __str__(self):
         return desc
 
     def _sector_and_sz_itr(self, elf_simple, data_start):
-        """Iterator which returns starting address and sector size"""
+        """Iterator which returns starting address and sector size
+           Also generates a list of all sector starting addresses followed by size
+        """
         for entry_start in count(data_start, self.FLASH_SECTORS_STRUCT_SIZE):
             data = elf_simple.read(entry_start, self.FLASH_SECTORS_STRUCT_SIZE)
             size, start = struct.unpack(self.FLASH_SECTORS_STRUCT, data)
             start_and_size = start, size
+            if len(self.all_sectors) > 0:
+                # Find the previous sector start and size
+                prev_start, prev_size = self.all_sectors[-1]
+                # Fill in all sectors between the previously listed sector up to the current sector
+                for new_start in range(prev_start+prev_size, start, prev_size):
+                    if not((new_start+prev_size) > self.size):
+                        self.all_sectors.append((new_start, prev_size))
+                    else:
+                        break
```

Comment:
The field `sector_info_list` should already have this info.

Reply:
`sector_info_list` will only keep one pair of (start address, size) for each equally sized sector.

Reply:
This bit will output the sectors from `0x00000400 - flash size`